client/src/search.tsx

```diff
@@ -13,14 +13,44 @@ function isMobileUserAgent() {
   );
 }
 
-export function SearchWidget() {
+function useFocusOnSlash(input: HTMLInputElement | null) {
+  useEffect(() => {
+    function focusOnSearchMaybe(event) {
+      // Don't do this if the current event target is a widget
+      if (
+        event.code === "Slash" &&
+        !["TEXTAREA", "INPUT"].includes(event.target.tagName)
+      ) {
+        if (input && document.activeElement !== input) {
+          event.preventDefault();
+          input.focus();
+        }
+      }
+    }
+    document.addEventListener("keydown", focusOnSearchMaybe);
+    return () => {
+      document.removeEventListener("keydown", focusOnSearchMaybe);
+    };
+  }, [input]);
+}
+
+export function SearchNavigateWidget() {
   const { pathname } = useLocation();
   const navigate = useNavigate();
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [query, setQuery] = useState("");
+
+  useFocusOnSlash(inputRef.current);
```

Comment:
pulled that out, to make the widget usable without it having to globally capture that key